### PR TITLE
add kcl lsp

### PR DESCRIPTION
**Pull request by xxmao123**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

re #11 
#### 2. What is the scope of this PR (e.g. component or file name):

src/main/java/io/kusionstack/kcl

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [x] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -30,7 +30,7 @@ dependencies {
 
 // Set the JVM language level used to build the project. Use Java 11 for 2020.3+, and Java 17 for 2022.2+.
 kotlin {
-    jvmToolchain(11)
+    jvmToolchain(17)
```
Comment (Peefy) [inline build.gradle.kts:33]: Pls add more PR information

File: build.gradle.kts [hunk 2]
```diff
@@ -30,7 +30,7 @@ dependencies {
 
 // Set the JVM language level used to build the project. Use Java 11 for 2020.3+, and Java 17 for 2022.2+.
```
Comment (Peefy) [inline build.gradle.kts:31]: The comment here use Java11 for 2020.3+, why change to Java17 here

Comment (xxmao123) [reply build.gradle.kts:31]: The comment here use Java 17 for 2022.2+.

**Comment by amyXia1994** [comment]
good job! Please upload the plugin distribution file to the PR so we could test it end to end

**Comment by xxmao123** [comment]
[intellij-kcl-0.1.15-lsp.zip](https://github.com/kcl-lang/intellij-kcl/files/12503185/intellij-kcl-0.1.15-lsp.zip)


> good job! Please upload the plugin distribution file to the PR so we could test it end to end

**Review by Peefy — APPROVED** [review]
LGTM

**Comment by amyXia1994** [comment]
Hi, mao, I tried install the plugin but faced the following error:

image


My IDEA version:
IntelliJ IDEA 2023.2.1 (Ultimate Edition)

> [intellij-kcl-0.1.15-lsp.zip](https://github.com/kcl-lang/intellij-kcl/files/12503185/intellij-kcl-0.1.15-lsp.zip)
> 
> > good job! Please upload the plugin distribution file to the PR so we could test it end to end

**Comment by xxmao123** [comment]
> Hi, mao, I tried install the plugin but faced the following error:
> 
> image
> 
> My IDEA version: IntelliJ IDEA 2023.2.1 (Ultimate Edition)
> 
> > [intellij-kcl-0.1.15-lsp.zip](https://github.com/kcl-lang/intellij-kcl/files/12503185/intellij-kcl-0.1.15-lsp.zip)
> > > good job! Please upload the plugin distribution file to the PR so we could test it end to end
[intellij-kcl.zip](https://github.com/kcl-lang/intellij-kcl/files/12511877/intellij-kcl.zip)

Sorry. Please try this zip again.

**Comment by amyXia1994** [comment]
> > Hi, mao, I tried install the plugin but faced the following error:
> > image
> > My IDEA version: IntelliJ IDEA 2023.2.1 (Ultimate Edition)
> > > [intellij-kcl-0.1.15-lsp.zip](https://github.com/kcl-lang/intellij-kcl/files/12503185/intellij-kcl-0.1.15-lsp.zip)
> > > > good job! Please upload the plugin distribution file to the PR so we could test it end to end
> 
> Sorry. Please try this zip again. [intellij-kcl.zip](https://github.com/kcl-lang/intellij-kcl/files/12511896/intellij-kcl.zip)

This zip works 👍  I configured the kcl-language-server in my PATH and now I get the auto completion provided from language server

But, cc @He1pa here are also some back-offs, such as the go-to-definition of the import statement.
The go-to action from the import path to the kcl file/directories works without lsp, but with lsp support the jump failed.

go-to-def-intellij

**Comment by amyXia1994** [comment]
I'll merge this PR and the problems mentioned [here](https://github.com/kcl-lang/intellij-kcl/pull/12#issuecomment-1716908447) will be discussed in this issue #13

**Pull request merged**